src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_scope_NES.c

```diff
@@ -115,137 +250,769 @@ u8 D_8082B924[] = {
+                sCursorEnvB = sCursorEnvColorTarget[pauseCtx->cursorColorSet + sCursorColorTargetIndex][2];
+                sCursorColorTargetIndex ^= 1;
+                sCursorColorTimer = 10;
+            }
+        }
+
+        if ((pauseCtx->pageIndex != PAUSE_ITEM) && (pauseCtx->pageIndex != PAUSE_QUEST)) {
+            gDPPipeSync(POLY_OPA_DISP++);
+
+            gDPSetCombineLERP(POLY_OPA_DISP++, TEXEL0, 0, PRIMITIVE, 0, TEXEL0, 0, SHADE, 0, TEXEL0, 0, PRIMITIVE, 0,
+                              TEXEL0, 0, SHADE, 0);
+
+            gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 180, 180, 120, 255);
+
+            Matrix_RotateYF(0.0f, MTXMODE_NEW);
+            Matrix_Translate(0.0f, sPauseMenuVerticalOffset / 100.0f, -93.0f, MTXMODE_APPLY);
+            Matrix_Scale(0.78f, 0.78f, 0.78f, MTXMODE_APPLY);
+            Matrix_RotateXFApply(-pauseCtx->itemPageRoll / 100.0f);
+
+            gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+            POLY_OPA_DISP = KaleidoScope_DrawPageSections(POLY_OPA_DISP, pauseCtx->itemPageVtx, sItemPageBgTextures);
+
+            KaleidoScope_DrawItemSelect(play);
+        }
+
+        if ((pauseCtx->pageIndex != PAUSE_MAP) && (pauseCtx->pageIndex != PAUSE_MASK)) {
+            gDPPipeSync(POLY_OPA_DISP++);
+
+            gDPSetCombineLERP(POLY_OPA_DISP++, TEXEL0, 0, PRIMITIVE, 0, TEXEL0, 0, SHADE, 0, TEXEL0, 0, PRIMITIVE, 0,
+                              TEXEL0, 0, SHADE, 0);
+
+            gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 180, 180, 120, 255);
+
+            Matrix_RotateYF(-1.57f, MTXMODE_NEW);
+            Matrix_Translate(0.0f, sPauseMenuVerticalOffset / 100.0f, -93.0f, MTXMODE_APPLY);
+            Matrix_Scale(0.78f, 0.78f, 0.78f, MTXMODE_APPLY);
+            Matrix_RotateXFApply(-pauseCtx->mapPageRoll / 100.0f);
+
+            gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+            POLY_OPA_DISP = KaleidoScope_DrawPageSections(POLY_OPA_DISP, pauseCtx->mapPageVtx, sMapPageBgTextures);
+
+            if (sInDungeonScene) {
+                KaleidoScope_DrawDungeonMap(play);
+                func_8012C8AC(gfxCtx);
+                gDPSetCombineMode(POLY_OPA_DISP++, G_CC_MODULATEIA_PRIM, G_CC_MODULATEIA_PRIM);
+                func_801091F0(play);
+            } else {
+                KaleidoScope_DrawWorldMap(play);
+            }
+        }
+
+        if ((pauseCtx->pageIndex != PAUSE_QUEST) && (pauseCtx->pageIndex != PAUSE_ITEM)) {
+            gDPPipeSync(POLY_OPA_DISP++);
+
+            gDPSetTextureFilter(POLY_OPA_DISP++, G_TF_BILERP);
+
+            gDPSetCombineLERP(POLY_OPA_DISP++, TEXEL0, 0, PRIMITIVE, 0, TEXEL0, 0, SHADE, 0, TEXEL0, 0, PRIMITIVE, 0,
+                              TEXEL0, 0, SHADE, 0);
+
+            gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 180, 180, 120, 255);
+
+            Matrix_RotateYF(-3.14f, MTXMODE_NEW);
```

Comment:
There might be enough instances of `3.14` and `1.57` to justify a define: `M_PI_APPOX 3.14`?

Reply:
I'm starting to think that, yeah. Although I'm concerned that there are probably a couple of other slightly better approximations.

Reply:
it's a weird thing because I don't think you can just swap out kaleido's pi approximations with M_PI
some timers are based on angles reaching exact float values (:YEP:) which wouldn't play well with many decimal places (or even few really but oh well...)
For example here in OoT
https://github.com/Dragorn421/oot/blob/7af42985adb19dcf4f4de33c6d66bc0d5949ee0e/src/overlays/misc/ovl_kaleido_scope/z_kaleido_scope_PAL.c#L3854
`if (pauseCtx->itemPageRoll == 160.0f)`
It may not look like it but this 160.0f is pi/2*100

Reply:
I wasn't expecting that to be possible, I don't understand what the relevance of that is, beyond that 160.0f and 314.0f are also special values related to pi.

Reply:
relevance of what regarding to what?

`itemPageRoll/100` is used as radians here if that's what you want to know? https://github.com/Dragorn421/oot/blob/7af42985adb19dcf4f4de33c6d66bc0d5949ee0e/src/overlays/misc/ovl_kaleido_scope/z_kaleido_scope_PAL.c#L1146

Reply:
But the values we're discussing in this PR are 3.14 and 1.57, which is not the same as 1.60.

Reply:
OK. I'm sorry to interrupt

Reply:
This does at least answer the question about there being other different values used, so I think we're going to have to comment on them for now and look at them in later PR when we have all the interface stuff in.

Reply:
I'm not really seeing what's being debated here. I think it's fair to say 1.6 radians is roughly 1.57 radius which is roughly pi / 2. I think it'd be good to tabulate all the "roughly equal to a multiple of pi" values and see how many could be expressed with `M_PI_APPOX 3.14` and which ones could not, but that's getting a bit outside the scope of this PR. So I think I'll leave it as constants for now